src/go/writer/writer.go

```diff
@@ -34,19 +37,39 @@ func (f *featureItem) NodeItem() index.NodeItem {
 	return *f.nodeItem
 }
 
+// writerOption is the additional option to be passed into Writer.
+type writerOption func(*Writer)
+
+// Use memory instead of temporary file when generating index
+// or execute HeaderUpdater.
+// Effective only when includeIndex is true or when HeaderUpdater is provided.
+func WithMemory() writerOption {
+	return func(w *Writer) {
+		w.useMemory = true
+	}
+}
+
 // NewWriter returns a new writer instance that will write a flatgeobuf file
 // with the given Header, a possible Index (depending on includeIndex), a
 // FeatureGenerator that will provide the Features to be written and a
 // HeaderUpdater that will be used to update the Header after all features
 // have been generated.
 func NewWriter(header *Header, includeIndex bool,
```

Comment:
Can you please add a new constructor like function instead? It will be better if NewWriter keeps working as before as this is a breaking change to the current API and it is easy enough to add a new one. Something like:

func NewWriterWithOptions(featureGenerator FeatureGenerator, headerUpdater HeaderUpdater, opts ...writerOption) *Writer {
  // Your implementation.
}

func NewWriter(featureGenerator FeatureGenerator, headerUpdater HeaderUpdater) *Writer {
  return NewWriterWithOptions(featureGenerator, headerUpdater, nil)
}

Reply:
@bruno-ga thanks for the review. 

The main reason that I am using this option pattern `opts ...writerOption` is to avoid breaking change. As shown in L104 of src/go/flatgeobuf_test.go in PR, the user can call the NewWriter function without passing `writer.WithMemory()` and retain the original behavior.

Based on the above, do you still think we need another construtor?

Reply:
Sorry, you are correct and this is an aspect of go variadic functions that I have not explored recently and kinda forgot about it. The parameter can, in fact, be omitted so this is fine.